### PR TITLE
issue #936 add service load mechanism for TM

### PR DESCRIPTION
**Pull request by sharajava**
### Ⅰ. Describe what this PR did

issue #936 add service load mechanism for TM

### CODE REVIEW

File: tm/src/main/java/io/seata/tm/TransactionManagerHolder.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    private static class SingletonHolder {
+
+        private static TransactionManager INSTANCE = null;
+
+        static {
+            try {
+                INSTANCE = EnhancedServiceLoader.load(TransactionManager.class);
+                LOGGER.info("TransactionManager Singleton " + INSTANCE);
+            } catch (Throwable anyEx) {
+                LOGGER.error("Failed to load TransactionManager Singleton! ", anyEx);
```
Comment (slievrly) [inline tm/src/main/java/io/seata/tm/TransactionManagerHolder.java]: should throw exx?

**Comment by CoffeeLatte007** [comment]
Who needs custom TransactionManager？Can you describe the usage scenarios？

**Comment by sharajava** [comment]
> Who needs custom TransactionManager？Can you describe the usage scenarios？

TM and RM are core interfaces in Seata, I think it's better to take them forward to transaction standard, not just Java SDK. Implementations of TM and RM should be the endpoint for service providers of Seata standard. I hope this change will be the first step of making Seata as a standard.

Native Seata

Seata Service Provider

File: tm/src/main/java/io/seata/tm/TransactionManagerHolder.java (hunk 1 above)
Comment (sharajava) [reply tm/src/main/java/io/seata/tm/TransactionManagerHolder.java]: Not sure, I was just considering that exception thrown from static block will cause some error of class initialization. 

Reference: https://stackoverflow.com/questions/2070293/why-doesnt-java-allow-to-throw-a-checked-exception-from-static-initialization-b

File: tm/src/main/java/io/seata/tm/TransactionManagerHolder.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        }
+    }
+
+    /**
+     * Get transaction manager.
+     *
+     * @return the transaction manager
+     */
+    public static TransactionManager get() {
+        return SingletonHolder.INSTANCE;
```
Comment (wgs13579) [inline tm/src/main/java/io/seata/tm/TransactionManagerHolder.java]: is there need NULL check ?

Comment (sharajava) [reply tm/src/main/java/io/seata/tm/TransactionManagerHolder.java]: Yes, I will add some check here.

File: tm/src/main/java/io/seata/tm/TransactionManagerHolder.java (hunk 1 above)
Comment (CoffeeLatte007) [reply tm/src/main/java/io/seata/tm/TransactionManagerHolder.java]: maybe catch check exception EnhancedServiceNotFoundException. other excepiton should throw?

**Pull request merged**